### ISSUE TITLE
Data Race in Compare

### ISSUE THREAD

**Issue by jxsl13** [pb33f/libopenapi#156]
Please test if you can reproduce.

```
=== RUN   TestDocument_Paths_As_Array
--- PASS: TestDocument_Paths_As_Array (0.00s)
=== RUN   TestNewDocument_terminable_circular_references
--- PASS: TestNewDocument_terminable_circular_references (0.00s)
=== RUN   TestLoadDocument_Simple_V2
--- PASS: TestLoadDocument_Simple_V2 (0.00s)
=== RUN   TestLoadDocument_Simple_V2_Error
--- PASS: TestLoadDocument_Simple_V2_Error (0.00s)
=== RUN   TestLoadDocument_Simple_V2_Error_BadSpec
--- PASS: TestLoadDocument_Simple_V2_Error_BadSpec (0.00s)
=== RUN   TestLoadDocument_Simple_V3_Error
--- PASS: TestLoadDocument_Simple_V3_Error (0.00s)
=== RUN   TestLoadDocument_Error_V2NoSpec
--- PASS: TestLoadDocument_Error_V2NoSpec (0.00s)
=== RUN   TestLoadDocument_Error_V3NoSpec
--- PASS: TestLoadDocument_Error_V3NoSpec (0.00s)
=== RUN   TestLoadDocument_Empty
--- PASS: TestLoadDocument_Empty (0.00s)
=== RUN   TestLoadDocument_Simple_V3
--- PASS: TestLoadDocument_Simple_V3 (0.00s)
=== RUN   TestLoadDocument_Simple_V3_Error_BadSpec
--- PASS: TestLoadDocument_Simple_V3_Error_BadSpec (0.00s)
=== RUN   TestDocument_Serialize_Error
--- PASS: TestDocument_Serialize_Error (0.00s)
=== RUN   TestDocument_Serialize
--- PASS: TestDocument_Serialize (0.00s)
=== RUN   TestDocument_Serialize_Modified
--- PASS: TestDocument_Serialize_Modified (0.00s)
=== RUN   TestDocument_RenderAndReload_ChangeCheck_Burgershop
--- PASS: TestDocument_RenderAndReload_ChangeCheck_Burgershop (0.17s)
=== RUN   TestDocument_RenderAndReload_ChangeCheck_Stripe
--- PASS: TestDocument_RenderAndReload_ChangeCheck_Stripe (20.73s)
=== RUN   TestDocument_RenderAndReload_ChangeCheck_Asana
--- PASS: TestDocument_RenderAndReload_ChangeCheck_Asana (2.90s)
=== RUN   TestDocument_RenderAndReload
--- PASS: TestDocument_RenderAndReload (0.29s)
=== RUN   TestDocument_RenderAndReload_Swagger
--- PASS: TestDocument_RenderAndReload_Swagger (0.08s)
=== RUN   TestDocument_BuildModelPreBuild
--- PASS: TestDocument_BuildModelPreBuild (0.29s)
=== RUN   TestDocument_AnyDoc
--- PASS: TestDocument_AnyDoc (0.00s)
=== RUN   TestDocument_AnyDocWithConfig
--- PASS: TestDocument_AnyDocWithConfig (0.00s)
=== RUN   TestDocument_BuildModelCircular
--- PASS: TestDocument_BuildModelCircular (0.01s)
=== RUN   TestDocument_BuildModelBad
--- PASS: TestDocument_BuildModelBad (0.03s)
=== RUN   TestDocument_Serialize_JSON_Modified
--- PASS: TestDocument_Serialize_JSON_Modified (0.00s)
=== RUN   TestExtractReference
param1: I am a param, is reference? true, original reference #/components/parameters/Param1--- PASS: TestExtractReference (0.00s)
=== RUN   TestDocument_BuildModel_CompareDocsV3_LeftError
--- PASS: TestDocument_BuildModel_CompareDocsV3_LeftError (0.10s)
=== RUN   TestDocument_BuildModel_CompareDocsV3_RightError
--- PASS: TestDocument_BuildModel_CompareDocsV3_RightError (0.10s)
=== RUN   TestDocument_BuildModel_CompareDocsV2_Error
--- PASS: TestDocument_BuildModel_CompareDocsV2_Error (0.17s)
=== RUN   TestDocument_BuildModel_CompareDocsV2V3Mix_Error
--- PASS: TestDocument_BuildModel_CompareDocsV2V3Mix_Error (0.18s)
=== RUN   TestSchemaRefIsFollowed
--- PASS: TestSchemaRefIsFollowed (0.00s)
=== RUN   TestDocument_ParamsAndRefsRender
--- PASS: TestDocument_ParamsAndRefsRender (0.01s)
=== RUN   TestDocument_ExampleMap
--- PASS: TestDocument_ExampleMap (0.00s)
=== RUN   TestDocument_OperationsAsRefs
--- PASS: TestDocument_OperationsAsRefs (0.00s)
=== RUN   TestDocument_InputAsJSON
--- PASS: TestDocument_InputAsJSON (0.00s)
=== RUN   TestDocument_InputAsJSON_LargeIndent
--- PASS: TestDocument_InputAsJSON_LargeIndent (0.00s)
=== RUN   TestDocument_RenderWithIndention
--- PASS: TestDocument_RenderWithIndention (0.00s)
=== RUN   ExampleNewDocument_fromOpenAPI3Document
--- PASS: ExampleNewDocument_fromOpenAPI3Document (0.11s)
=== RUN   ExampleNewDocument_fromWithDocumentConfigurationFailure
--- PASS: ExampleNewDocument_fromWithDocumentConfigurationFailure (0.13s)
=== RUN   ExampleNewDocument_fromWithDocumentConfigurationSuccess
--- PASS: ExampleNewDocument_fromWithDocumentConfigurationSuccess (18.25s)
=== RUN   ExampleNewDocument_fromSwaggerDocument
--- PASS: ExampleNewDocument_fromSwaggerDocument (0.08s)
=== RUN   ExampleNewDocument_fromUnknownVersion
--- PASS: ExampleNewDocument_fromUnknownVersion (0.06s)
=== RUN   ExampleNewDocument_mutateValuesAndSerialize
--- PASS: ExampleNewDocument_mutateValuesAndSerialize (0.00s)
=== RUN   ExampleCompareDocuments_openAPI
==================
WARNING: DATA RACE
Read at 0x00c02d547da0 by goroutine 259619:
  github.com/pb33f/libopenapi/what-changed/model.CreateChange()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:48 +0x438
  github.com/pb33f/libopenapi/what-changed/model.checkRightValue[...]()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:352 +0x2e8
  github.com/pb33f/libopenapi/what-changed/model.CheckMapForChangesWithComp[...].func3()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:329 +0xa4

Previous write at 0x00c02d547da0 by goroutine 259590:
  github.com/pb33f/libopenapi/what-changed/model.CreateChange()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:48 +0x4d0
  github.com/pb33f/libopenapi/what-changed/model.checkRightValue[...]()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:352 +0x2e8
  github.com/pb33f/libopenapi/what-changed/model.CheckMapForChangesWithComp[...].func3()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:329 +0xa4

Goroutine 259619 (running) created at:
  github.com/pb33f/libopenapi/what-changed/model.CheckMapForChangesWithComp[...]()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:329 +0x8bc
  github.com/pb33f/libopenapi/what-changed/model.CheckMapForAdditionRemoval[...]()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:245 +0xc4
  github.com/pb33f/libopenapi/what-changed/model.runComparison[...]()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/components.go:233 +0x1f0
  github.com/pb33f/libopenapi/what-changed/model.CompareComponents.func2()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/components.go:137 +0x78

Goroutine 259590 (finished) created at:
  github.com/pb33f/libopenapi/what-changed/model.CheckMapForChangesWithComp[...]()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:329 +0x8bc
  github.com/pb33f/libopenapi/what-changed/model.CheckMapForAdditionRemoval[...]()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:245 +0xc4
  github.com/pb33f/libopenapi/what-changed/model.runComparison[...]()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/components.go:233 +0x1f0
  github.com/pb33f/libopenapi/what-changed/model.CompareComponents.func4()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/components.go:149 +0x78
==================
==================
WARNING: DATA RACE
Read at 0x00c0069bc9b8 by goroutine 259619:
  runtime.growslice()
      /opt/homebrew/Cellar/go/1.20.7/libexec/src/runtime/slice.go:157 +0x0
  github.com/pb33f/libopenapi/what-changed/model.CreateChange()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:48 +0x470
  github.com/pb33f/libopenapi/what-changed/model.checkRightValue[...]()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:352 +0x2e8
  github.com/pb33f/libopenapi/what-changed/model.CheckMapForChangesWithComp[...].func3()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:329 +0xa4

Previous write at 0x00c0069bc9b8 by goroutine 259590:
  github.com/pb33f/libopenapi/what-changed/model.CreateChange()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:48 +0x498
  github.com/pb33f/libopenapi/what-changed/model.checkRightValue[...]()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:352 +0x2e8
  github.com/pb33f/libopenapi/what-changed/model.CheckMapForChangesWithComp[...].func3()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:329 +0xa4

Goroutine 259619 (running) created at:
  github.com/pb33f/libopenapi/what-changed/model.CheckMapForChangesWithComp[...]()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:329 +0x8bc
  github.com/pb33f/libopenapi/what-changed/model.CheckMapForAdditionRemoval[...]()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:245 +0xc4
  github.com/pb33f/libopenapi/what-changed/model.runComparison[...]()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/components.go:233 +0x1f0
  github.com/pb33f/libopenapi/what-changed/model.CompareComponents.func2()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/components.go:137 +0x78

Goroutine 259590 (finished) created at:
  github.com/pb33f/libopenapi/what-changed/model.CheckMapForChangesWithComp[...]()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:329 +0x8bc
  github.com/pb33f/libopenapi/what-changed/model.CheckMapForAdditionRemoval[...]()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/comparison_functions.go:245 +0xc4
  github.com/pb33f/libopenapi/what-changed/model.runComparison[...]()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/components.go:233 +0x1f0
  github.com/pb33f/libopenapi/what-changed/model.CompareComponents.func4()
      /Users/john/Desktop/Development/libopenapi/what-changed/model/components.go:149 +0x78
==================
--- PASS: ExampleCompareDocuments_openAPI (0.16s)
=== RUN   ExampleCompareDocuments_swagger
--- PASS: ExampleCompareDocuments_swagger (0.21s)
=== RUN   ExampleNewDocument_infinite_circular_references
--- PASS: ExampleNewDocument_infinite_circular_references (0.00s)
=== RUN   ExampleNewDocument_unpacking_extensions
--- PASS: ExampleNewDocument_unpacking_extensions (0.00s)
=== RUN   ExampleNewDocument_modifyAndReRender
--- PASS: ExampleNewDocument_modifyAndReRender (0.28s)
FAIL
	github.com/pb33f/libopenapi	coverage: 98.1% of statements
FAIL	github.com/pb33f/libopenapi	44.818s
```

**Comment by daveshanley** [comment]
I will double check I ran a race check a few months back without any issues - but I will re-check

**Comment by jasonquigleyhpecom** [comment]
Running `go test -race ./...` on this repo results in a huge number of races :-(

**Comment by daveshanley** [comment]
Oh dear. 

This wasn't the case a few months back. I will spend some time debugging this. However it may be worth waiting, we have a huge refs for of async code coming in #148

**Comment by daveshanley** [comment]
Ok, there are a ton of these issues that are simple to fix. Will include with my latest PR.

**Comment by daveshanley** [comment]
This will be fixed in #162

**Comment by daveshanley** [comment]
Fixed in `v0.10.1`
